logger/logSubsystem.go

```diff
@@ -0,0 +1,110 @@
+package logger
+
+import (
+	"encoding/hex"
+	"io"
+	"os"
+	"strings"
+	"sync"
+)
+
+const ellipsisCharacter = "\u2026"
+
+var logMut = &sync.RWMutex{}
+var loggers map[string]*logger
+var defaultLogOut LogOutputHandler
+var defaultLogLevel = LogInfo
+
+func init() {
+	logMut.Lock()
+	defer logMut.Unlock()
```

Comment:
remove `defer` and move this after the protected lines

Reply:
done